enhancements/machine-config/manage-boot-images.md

```diff
@@ -472,24 +645,112 @@ status:
 
-How will security be reviewed and by whom? TBD
+How will security be reviewed and by whom?
 This is a solution aimed at reducing usage of outdated artifacts and should not introduce any security concerns that do not currently exist. 
 
-How will UX be reviewed and by whom? TBD 
+How will UX be reviewed and by whom? 
 The UX element involved include the user opt-in and opt-out, which is currently up for debate. 
 
+#### Enabling Boot Image Updates by default
+
+This will be done on a platform by platform basis. Some key benchmarks have to be met for a platform to be considered ready
+for default on:
+- Sufficent runtime(say, at least 1 release) has been accumulated while boot image updates has been GAed for this platform.
+- Periodic tests have been added for this platform in CI and have met certain passing metrics. This should include Y stream upgrade e2es.
+- Any teams that are affected by default-on
+behavior have been notified and assisted with the transition.
+
+The default-on flow could look like this:
+```mermaid
+flowchart-elk LR;
+    Start((Start)) -->PlatformCheck[Does the cluster platform have boot image updates suppport?]
+    PlatformCheck -->|Yes|MachineConfigurationCheck[Does the cluster currently have a boot image update configuration?]
+    PlatformCheck -->|No| Stop ;
+    MachineConfigurationCheck -->|Yes|Stop
+    MachineConfigurationCheck -->|No| UpdateConfig["The MCO will inject a default configuration based on the platform, which could be opt-in or opt-out"] ;    
+    UpdateConfig --> Stop((Stop));
+```
+Some points to note:
+- For bookkeeping purposes, the MCO will annotate the `MachineConfiguration` object when opting in the cluster by default.
+- This mechanism will be active on installs and upgrades. 
+- If the cluster admin wishes to opt-out of the feature, they have to do so by explicitly opting out the cluster via the API knob prior to the upgrade. 
+- If any of the MachineSets have an OwnerReference, it will be skipped for boot image updates. This will cause an alert/warning to the cluster admin, but it will no longer cause a degrade. 
+
+
+### Enforcement of bootimage skew
+
+There should be some mechanism that will alert the user when a cluster's bootimage are out of date. To allow for this, the `coreos-bootimages` configmap will gain a new field, which will store the boot image required for upgrading to the next y stream release.
+
+Generally speaking, we would like to keep the minimum required bootimage version aligned to the RHEL version we are shipping in the payload and constant for a given release's z stream. For example, a 9.6 bootimage will be allowed until 9.8 is shipped via RHCOS. We would like to keep this customizable, such that any major breaking changes outside of RHEL major/minor can still be enforced as a one-off.
+
+#### Enforcement options
```

Comment:
Do we need to make a choice between the options, or are they all going to be implemented?

Reply:
Ack, and it's all going to be feature gated for now?